### PR TITLE
46631448 - Siamese ADNI Model

### PR DESCRIPTION
**Pull request by LeSebring**
Hello COMP3710 Teaching Team,

I am submitting a Pull Request for the Project assignment. In my project, I attempted to solve task 7, that is to create a classifier based on a Siamese model on the ADNI dataset. Unfortunately, I was only able to achieve around 50% accuracy with my model, which is akin to random guessing, so a classifier based on my model would perform very poorly. This may be due to flaws in my model or lack of training.

Please refer to the README.md in my pull request (the one inside the /recognition/ folder) for more information on my project.

Regards,

Jason Alexander Lukas, 46631448

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Poor
- **embedded images are not showing up as they haven't been included in the repo**
- only describes what code is included in each of the files
- could have used diagrams to explain model architectures
- no information on how the siamese+classifier architecture works
- no loss/accuracy plots, only final numbers provided

Commit messages: OK

Code: 
- poor design
- **siamese architecture is incorrect**. Network outputs only a probablility and a wrong loss function used
- although required 4 python files are present, code components within them are not planned well and included somewhat incorrectly
- generating pairs of images seem to be static and fixed
- sufficient comments

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification of individual images**
- siamese calidation accuracy is ~50%, probably because the model is incorrectly implemented

General comments:
- **Algorithm does not solve the respective pattern recognition problem appropriately**
- does not seem to have understood how the network needs to be applied for classification

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, no classification of single images -3
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, but no output provided, can't verify -2, poor performance < 0.6 DSC -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no plot or validation set plots -2
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK, but missing usage -1, result output -2
Model/technical explanation missing architecture etc. -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove pyc files to allow merge, restore repo README and remove gitignore file. -2
Request Description OK, could be better -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by LeSebring** [comment]
Hello @shakes76,
I have implemented your feedback by removing the .gitignore file, the .pyc files, and by restoring the repo's README.md file.
Thank you.
Sincerely,
Jason

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
